src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/Algebras/CharSetSolver.cs

```diff
@@ -40,10 +40,23 @@ public BDD CharConstraint(char c, bool ignoreCase = false, string? culture = nul
             else
             {
                 //individual character BDDs are always fixed
-                return _charPredTable[c] ??= CreateSetFrom(c, 15);
+                if (c < _charPredTable.Length)
+                    return _charPredTable[c] ??= CreateBDDFromChar(c);
+                else
+                    return CreateBDDFromChar(c);
```

Comment:
Storing the array from an instance field into a local enables better bounds-check elimination:
```suggestion
                BDD[] charPredTable = _charPredTable;
                return c < charPredTable ?
                    charPredTable[c] ??= CreateBDDFromChar(c) :
                    CreateBDDFromChar(c);
```